### DISCUSSION TITLE
Why is the number of CPUs on a VM limited to 128 on the host of Ampere Altra?

### DISCUSSION THREAD

**Discussion by peng6662001** [cloud-hypervisor/cloud-hypervisor#5331] [Q&A] [answered]
I use the command to start VM on Ampere Altra:
```
./cloud-hypervisor --kernel Image --disk path=../focal-server-cloudimg-arm64.raw --disk path=/tmp/ubuntu-cloudinit.img --cmdline 'root=/dev/vda1 rw systemd.journald.forward_to_console=1' --cpus boot=130,max=130 --memory size=2G --api-socket /tmp/ch0 --rng src=/dev/urandom -v
```
The cpu numbers are always 128 if I use larger number for boot and max.
How to modify this limit?

**Comment by michael2012z** [comment, marked as answer]
You can change the 128 limit by modifying the `CONFIG_NR_CPUS` in guest kernel configuration:

https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/resources/linux-config-aarch64#L383

**Reply by peng6662001** [reply]
Thank you.I have verified.
